variables.tf

```diff
@@ -141,52 +180,49 @@ variable "argo_info" {
 }
 
 variable "argo_sync_policy" {
+  type        = any
   description = "ArgoCD syncPolicy manifest parameter"
   default     = {}
 }
 
 variable "argo_metadata" {
-  default = {}
-  # example value:
-  # argo_metadata = {
-  #   "annotations" : {}
-  #   "labels" : {}
-  #   "finalizers" : [
-  #     "resources-finalizer.argocd.argoproj.io"
-  #   ]
-  # }
+  type = any
+  default = {
+    "finalizers" : [
+      "resources-finalizer.argocd.argoproj.io"
```

Comment:
This finalizer we dont want as default especially for argocd as I understood from @tomas-balaz post

Reply:
my bad during merge